src/server/request_handler/default_request_handler.ts

```diff
@@ -338,6 +338,38 @@ export class DefaultRequestHandler implements A2ARequestHandler {
         return task;
     }
 
+    async listTasks(
+        params: ListTasksParams
+    ): Promise<ListTasksResult> {
+
+        const filteredTasks = await this.taskStore.list(params);
+        filteredTasks.forEach(task => {
+            const historyLength = params.historyLength ?? 0;
+            task.history = historyLength > 0 ? task.history?.slice(-historyLength) : [];
+
+            if (!params.includeArtifacts && task.artifacts){
+                task.artifacts = []
+            }
+        })
```

Comment:
For task stores backed by DB, this may be done at DB level (i.e. joining history/artifacts tables for relational DBs), as store receives same params and has all the information for all filtering. Applying this logic on top of that will corrupt results for history for example.

I think it makes more sense to apply all the logic at store level, as it may take advantage of the underlying storage implementation and is the best place to do it.